libs/net/dal/Services/ContentService.cs

```diff
@@ -150,6 +160,249 @@ public IPaged<Content> Find(ContentFilter filter, bool asNoTracking = true)
         return new Paged<Content>(items, filter.Page, filter.Quantity, total);
     }
 
+    /// <summary>
+    /// Find content that matches the specified 'filter'.
+    /// </summary>
+    /// <param name="filter">Filter to apply to the query.</param>
+    /// <returns>A page of content items that match the filter.</returns>
+    public async Task<IPaged<Content>> FindAsync(ContentFilter filter)
+    {
+        var shouldQueries = new List<Func<QueryContainerDescriptor<Content>, QueryContainer>>();
+
+        foreach (var productId in filter.ProductIds)
+        {
+            shouldQueries.Add(s => s.Term(t => t.ProductId, productId));
+        }
+
+        foreach (var contentId in filter.ContentIds)
+        {
+            shouldQueries.Add(s => s.Term(t => t.Id, contentId));
+        }
+
+        var filterQueries = new List<Func<QueryContainerDescriptor<Content>, QueryContainer>>();
+
+        if (!string.IsNullOrWhiteSpace(filter.OtherSource))
+        {
+            filterQueries.Add(s => s.Term(t => t.OtherSource, filter.OtherSource.ToLower()));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.Headline))
+        {
+            filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Headline).Value($"*{filter.Headline.ToLower()}*")));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.PageName))
+        {
+            filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Page).Value($"*{filter.PageName.ToLower()}*")));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.Section))
+        {
+            filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Section).Value($"*{filter.Section.ToLower()}*")));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.Edition))
+        {
+            filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Edition).Value($"*{filter.Edition.ToLower()}*")));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.Product))
+        {
+            filterQueries.Add(s => s.Term(t => t.ProductId, filter.Product) || (
+                s.Exists(e => e.Field(p => p.Product)) &&
+                s.Wildcard(m => m.Field(p => p.Product!.Name).Value($"*{filter.Product.ToLower()}*"))
+                ));
+        }
+
+        if (!string.IsNullOrWhiteSpace(filter.Byline))
+        {
+            filterQueries.Add(s => s.Wildcard(m => m.Field(p => p.Byline).Value($"*{filter.Byline.ToLower()}*")));
+        }
+
+        if (filter.ContentType.HasValue)
+        {
+            filterQueries.Add(s => s.Match(m => m.Field(p => p.ContentType).Query(filter.ContentType.Value.ToString())));
+        }
+
+        if (!filter.IncludeHidden.HasValue)
+        {
+            filterQueries.Add(s => !s.Exists(e => e.Field(p => p.IsHidden)) || s.Term(t => t.IsHidden, false));
+        }
+        else
+        {
+            filterQueries.Add(s => s.Term(t => t.IsHidden, filter.IncludeHidden.Value));
+        }
+
+        if (filter.ProductId.HasValue)
```

Comment:
Do not duplicate the logic already found in the `ProductIds` array.  We only need the array to filter for products.

Reply:
Removed, thanks